src/front/app/features/challenge/challenge.actions.js

```diff
@@ -0,0 +1,18 @@
+import api from '../../common/api';
+export const GET_CHALLENGE = 'GET_CHALLENGE';
+
+function receiveChallenge(challenge) {
+    return {
+        type: GET_CHALLENGE,
+        challenge
+    };
+}
+
+export function fetchChallenge(id) {
+    return function innerFetchChallenge(dispatch) {
```

Comment:
the name innerFetchChallenge is not useful, it is already seen that this is inner function